cmd/soroban-rpc/internal/daemon/daemon.go

```diff
@@ -187,16 +192,54 @@ func MustNew(cfg config.LocalConfig) *Daemon {
 	}
 }
 
-func Run(cfg config.LocalConfig, endpoint string) {
+func Run(cfg config.LocalConfig, endpoint string, adminEndpoint string) {
 	d := MustNew(cfg)
-	supporthttp.Run(supporthttp.Config{
-		ListenAddr: endpoint,
-		Handler:    d,
-		OnStarting: func() {
-			d.logger.Infof("Starting Soroban JSON RPC server on %v", endpoint)
-		},
-		OnStopped: func() {
-			d.Close()
-		},
-	})
+
+	server := &http.Server{
+		Addr:        endpoint,
+		Handler:     d,
+		ReadTimeout: defaultReadTimeout,
+	}
+
+	d.logger.Infof("Starting Soroban JSON RPC server on %v", endpoint)
+
+	go func() {
+		if err := server.ListenAndServe(); !errors.Is(err, http.ErrServerClosed) {
+			// Error starting or closing listener:
+			d.logger.Fatalf("Soroban JSON RPC server encountered fatal error: %v", err)
+			os.Exit(1)
+		}
+	}()
+	var adminServer *http.Server
+	if adminEndpoint != "" {
+		// after importing net/http/pprof, debug endpoints are implicitly registered in the default serve mux
+		adminServer = &http.Server{Addr: adminEndpoint, Handler: http.DefaultServeMux}
```

Comment:
Should we support adminEndpoint == endpoint ? ( it's a bit more complicated then it sounds )

Reply:
No, we shouldn't. There will be a binding error if both endpoints are the same.

Reply:
yes, that's why I said it's a little bit more complicated. ( i.e. in this case, we don't need another listener )

Reply:
but I'm not really picky about this one. Just figured it's the "complementary" to ensure we support all end user permutations. in particular, avoid generating binding error if the two endpoints are the same.